recipe_scrapers/nihhealthyeating.py

```diff
@@ -1,8 +1,18 @@
 # mypy: disallow_untyped_defs=False
+from attr import dataclass
 from ._abstract import AbstractScraper
 from ._exceptions import ElementNotFoundInHtml
 from ._utils import get_minutes, get_yields, normalize_string
 
+import re
+from bs4 import Tag
+
+
+@dataclass
+class IngredientGroup:
+    ingredients: list[str]
+    subproduct: str = None  # the part of the recipe that these ingredients are used for - for example, "dipping sauce"
```

Comment:
I wonder whether `purpose` (in the sense of a-reason-for-something) would be better?

(example: 'this group of ingredients is for `{purpose}`' -- where `purpose` is a section heading ("For the sauce"), or a component of a recipe ("Side salad"))

It's not exactly a culinary term, but it also feels more human-friendly than `subproduct`.  And it allows a little bit more flexibility in interpretation for scraper developers, while still communicating the intended use of the field.